workspaces/spectacle/src/index.ts

```diff
@@ -216,13 +219,24 @@ export async function makeSpectacle(opticContext: IOpticContext) {
     return projections;
   }
 
-  opticContext.specRepository.notifications.on('change', () => {
-    console.count('reloading because of specRepository change');
-    reload(opticContext);
-  });
+  async function reloadFromSpecChange(
+    specChanges: AsyncGenerator<number>,
+    opticContext: IOpticContext
+  ) {
+    for await (let generation of specChanges) {
+      console.log('reloading because of specRepository change', generation);
+      await reload(opticContext);
+    }
+  }
 
   await reload(opticContext);
 
+  // TODO: make sure this Promise is consumed somewhere so errors are handled. Return from makeSpectacle perhaps?
```

Comment:
@devdoshi right now, in browsers and modern versions of node, errors here should cause a `UnhandledPromiseRejectionWarning`, so it _shouldn't_ get swallowed silently. Good enough for now, or should we bubble this up somewhere? I can imagine combining all async operations in a `Promise.all` and returning in a `result` promise.